src/protocol/sort/bit_permutation.rs

```diff
@@ -43,18 +52,21 @@ impl<'a, F: Field> BitPermutation<'a, F> {
     ///
     /// ## Errors
     /// It will propagate errors from multiplication protocol.
-    pub async fn execute(
-        &self,
-        ctx: ProtocolContext<'_, Replicated<F>, F>,
-    ) -> Result<Vec<Replicated<F>>, BoxError> {
-        let share_of_one = Replicated::one(ctx.role());
+    pub async fn execute(&self, ctx: ProtocolContext<'a, S, F>) -> Result<Vec<S>, BoxError>
+    where
+        ProtocolContext<'a, S, F>: SecureMul<F, Share = S>,
+    {
+        let share_of_one = S::one(
+            ctx.role(),
+            ctx.narrow(&ShareOfOne).prss().generate_replicated(RECORD_0),
```

Comment:
We don't want to generate a new `r`. We need to use the value that the ProtocolContext is already aware of. I recommend migrating all instances of:
`S::one(...)` to `ctx.share_of_one()`. The Context knows if it is malicious or not, and it knows what `r` is, so it can generate something of the form: 
`MaliciousReplicated::new(Replicated::one(self.role()), self.r_share)`

Reply:
Hmm its stored in SecurityValidator and not the context. However, context has prss and we can use it to get the r_share through prss.generate_replicated(RECORD_0) but this will make prss validation check fail (since we would end up requesting prss for RECORD_0 multiple times.